### DISCUSSION TITLE
Import not working or unknown file extension".ts"

### DISCUSSION THREAD

**Discussion by jcroma** [TypeStrong/ts-node#1524] [Q&A]
Hello there, I am trying to migrate on of our library to typescript. Decided to use ts-node since our tests are working on it.
Changed all files to `.ts` then import as node_modules library.
When I'm trying to use ` "type": "module"`  in package.json of library I'm getting error:
```
(node:1836) ExperimentalWarning: The ESM module loader is experimental.
TypeError [ERR_UNKNOWN_FILE_EXTENSION]: Unknown file extension ".ts" for [...path]
```
When I remove `type` field from package.json and launch then I get:
```
import * as program from 'commander';
^^^^^^

SyntaxError: Cannot use import statement outside a module
```
So non of most solution worked for me. 

My tsconfig.json:
```
    "ts-node": {
        "compilerOptions": {
            "module": "CommonJS",
            "esModuleInterop": true,
            "declaration": true,
        }
    }
```
execution of library:
```
"ts-node ./node_modules/qtest-integration/lib/qtest-main --rename"
```
node version: `v12.18.2`
typescript: `"version": "3.9.6"`

Any ideas what I'm doing wrong ?

**Comment by RossiniMaximo** [comment]
Hello , i was wondering if you fixed it because i'm having the same issue and i can't figure out what it is .

**Reply by RossiniMaximo** [reply]
It was "type": "module" giving the error , i don't know why because i put it there because it was supposedly required

**Reply by jcroma** [reply]
I was able to solve this issue. For my case it was problem with omitting node_modules due to ts-node basic behaviour.
Fix for that was:
add empty ignore in tsconfig.json:
```
    "ts-node": {
        "ignore": []
    }
```
and also by cli command: `ts-node --skip-ignore ./node_modules/your_package`
